underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsPositionedUnderFileInputStream.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.hdfs;
+
+import alluxio.underfs.SeekableUnderFileInputStream;
+import alluxio.util.io.BufferUtils;
+
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.fs.FSDataInputStream;
+
+import java.io.IOException;
+
+/**
+ * The input stream of HDFS as under filesystem. This input stream supports seeking but internally
+ * uses the positionedRead {@link FSDataInputStream} API. This stream can be cached for reuse.
+ */
+public class HdfsPositionedUnderFileInputStream extends SeekableUnderFileInputStream {
+  private long mPos;
+
+  HdfsPositionedUnderFileInputStream(FSDataInputStream in, long pos) {
+    super(in);
+    mPos = pos;
+  }
+
+  @Override
+  public int available() throws IOException {
+    if (mPos != ((FSDataInputStream) in).getPos()) {
+      return 0;
+    }
+    return in.available();
+  }
+
+  @Override
+  public long getPos() throws IOException {
+    return mPos;
+  }
+
+  @Override
+  public int read() throws IOException {
+    byte[] buffer = new byte[1];
+    int bytesRead = ((FSDataInputStream) in).read(mPos, buffer, 0, buffer.length);
+    if (bytesRead > 0) {
+      mPos += bytesRead;
```

Comment:
Consider using this class's `read(byte[])` method? Then we should only need to bookkeep `mPos` in one place

Reply:
thx, done